cmd/kubeadm/app/phases/upgrade/migrate.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"os"
+	"strings"
+
+	"github.com/coredns/deployment/kubernetes/migration"
+	"github.com/pkg/errors"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/dns"
+	"k8s.io/kubernetes/cmd/kubeadm/app/preflight"
+)
+
+// CoreDNSCheck validates installed kubelet version
+type CoreDNSCheck struct {
+	name   string
+	client clientset.Interface
+	f      func(clientset.Interface) error
+}
+
+func (c CoreDNSCheck) Name() string {
+	return c.name
+}
+
+func (c CoreDNSCheck) Check() (warnings, errors []error) {
+	if err := c.f(c.client); err != nil {
+		return nil, []error{err}
+	}
+	return nil, nil
+}
+
+// RunCoreDNSMigrationCheck initializes checks related to CoreDNS migration.
+func RunCoreDNSMigrationCheck(client clientset.Interface, ignorePreflightErrors sets.String) error {
+	currentDNSType, _, err := dns.DeployedDNSAddon(client)
```

Comment:
This one is also present in the `ClusterConfiguration` which is already fetched from the config map at this point.